source/draw/RasterMonochrome.ooc

```diff
@@ -148,13 +151,52 @@ RasterMonochrome: class extends RasterPacked {
 			source += sourceStride
 		}
 	}
+	_getFirstDerivative_optimized: func (region: IntBox2D, imageX, imageY: FloatImage) {
+		step := 2
+		sourceStride := this stride
+		destinationX := imageX pointer
+		destinationY := imageY pointer
 
+		// create small buffer to minimize cache misses for vertical pass
+		windowCacheWidth := region width + 4 * step
+		windowCacheHeight := region height + 4 * step
+		buffer := ByteBuffer new(windowCacheHeight * windowCacheWidth)
+		windowCache := buffer pointer as Byte*
+		for (y in 0 .. windowCacheHeight) {
+			lineIndex := region leftTop y + (y - 2 * step)
+			sourceY := this buffer pointer + lineIndex * sourceStride - step * 2 + region leftTop x
+			memcpy(windowCache + y * windowCacheWidth, sourceY, windowCacheWidth)
+		}
+
+		windowCache += step * 2 * windowCacheWidth
+		for (y in 0 .. region height) {
+			for (x in 2 * step .. region width + 2 * step) {
+				destinationX@ = (
+					8 * windowCache[x + step] -
+					8 * windowCache[x - step] +
+					windowCache[x - 2 * step] -
+					windowCache[x + 2 * step]
+				) as Float / (12.0f * step)
+				destinationX += 1
+
+				destinationY@ = (
+					8 * windowCache[x + windowCacheWidth * step] -
+					8 * windowCache[x - windowCacheWidth * step] +
+					windowCache[x - windowCacheWidth * 2 * step] -
+					windowCache[x + windowCacheWidth * 2 * step]
+				) as Float / (12.0f * step)
+				destinationY += 1
+			}
+			windowCache += windowCacheWidth
+		}
+
```

Comment:
Let's get rid of this empty line and the one on 170, the `}` line is enough space methinks.